discovery/swarm/swarm.go

```diff
@@ -0,0 +1,504 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package swarm
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/mwitkow/go-conntrack"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/prometheus/prometheus/util/strutil"
+	"github.com/prometheus/prometheus/util/yaml"
+)
+
+const (
+	// metaLabelPrefix is the meta prefix used for all meta labels in this discovery.
+	metaLabelPrefix = model.MetaLabelPrefix + "swarm_"
+	// serviceLabelPrefix is the prefix for the service labels.
+	serviceLabelPrefix = metaLabelPrefix + "service_label_"
+
+	// serviceLabel is used for the name of the service in Swarm.
+	serviceLabel model.LabelName = metaLabelPrefix + "service"
+	// imageLabel is the label that is used for the docker image running the service.
+	imageLabel model.LabelName = metaLabelPrefix + "image"
+	// taskLabel contains the task name of the app instance.
+	taskLabel model.LabelName = metaLabelPrefix + "task"
+	// nodeIPLabel contains the node ip.
+	nodeIPLabel model.LabelName = metaLabelPrefix + "node_ip"
+	// nodeNameLabel contains the node ip.
```

Comment:
typo ?  should be `// nodeNameLabel contains the node name.`

Reply:
Already fixed, thx.